docs/K8s-test-plan.md

```diff
@@ -0,0 +1,362 @@
+
+# Test Cases
+These test cases focus on Device Under Test (DUT) status changes in response to different configurations. {feature x} represents the single SONiC feature/container under test. 
+
+Each DUT has the following key configurations relevant to Kubernetes:
+- `disable` flag enables and disables the Kubernetes master. `disable=false `enables DUT to join Kubernetes master to allow Kubernetes management of SONiC features. `disable=true` triggers the DUT to reset from Kubernetes master, enforcing local management of all SONiC features
+- `set_owner` specifies desired owner for each feature- either local or kube
+- `ip` is the VIP of the high availability Kubernetes master. In these test cases, we assume that the VIP is configured properly
+
+Each DUT has the following key statuses relevant to Kubernetes: 
+- `connected` true means that the DUT has successfully joined to high availability Kubernetes master
+- `container_version` keeps track of which version is currently running for each feature
+- `remote_state` describes the state of Kubernetes-managed container (`stopped`, `running`, `pending`, or `none`) for each feature
+- `current_owner` describes the current owner for each feature- either local or kube
+
+## Test Scenario: Join/Reset
+
+### TC_JOIN_1: Join Master Once Available
+#### Test Objective
+Verify Device Under Test (DUT) joins high availability master once the VIP and Kubernetes API Server running on backend master servers become available.
+#### Test Configuration
+- All features running in local mode, with some having `set_owner=kube`
+- Running config = saved config_db.json
+- No manifests applied
+- VIP configured
+#### Test Steps 
+1. Set kube server `disable=true`
+   - **Expect:** kube server status shows `connected=false`
+2. Make VIP unreachable
+   - **Expect:** No change in kube server status, `connected=false`
+3. Stop API service in backend master servers
+   - **Expect:** No change in kube server status, `connected=false`
+4. Set kube server `disable=false`
+   - **Expect:** No change in kube server status, `connected=false`
+5. Make VIP reachable
+   - VIP becomes reachable but VIP::port is not available, as backend master servers are not running Kubernetes API Server
+   - **Expect:** No change in kube server status, `connected=false`
+6. Start API service in backend master servers
+   - VIP::port and VIP should both be available
+   - **Expect:** kube server status shows `connected=true`
+
+### TC_JOIN_2: Test Disable Flag
+#### Test Objective
+Verify Device Under Test (DUT) responds appropriately to kube server `disable` flag by joining master when `disable=false` and resetting from master when `disable=true`.
+#### Test Configuration
+- Pick up from step 6 of [TC_JOIN_1](#tc_join_1-join-master-once-available)
```

Comment:
Describing test configuration like this makes the test cases dependent on other test cases. It would be more clear if the test cases can be relatively independent. Can you consider describing the preconditions of each test in a more independent way? Like briefly describe what is the required state and configuration to start each test.

Reply:
I see why this description format makes it seem like some test cases depend on other test cases. I described it with "pick up from {another test case}" to make the test steps less wordy/redundant, and focus on what is being tested in the test case. In the actual test code, there will be no dependence on other test cases. 
 
Based on your feedback, I changed the `Test Configuration` header to `Test Setup` and describe the desired status and configuration of the DUT, features, Kubernetes server. 

To keep the test steps more concise, is it okay to say "To get to desired Test Setup state, pick up from {another test case}" after describing the required status and configuration for Test Setup?